pyannote/audio/interactive/diff/recipe.py

```diff
@@ -0,0 +1,239 @@
+import os
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Union
+
+import prodigy
+from prodigy import set_hashes
+from prodigy.components.loaders import Audio as AudioLoader
+from pyannote.core import Annotation, Segment
+from pyannote.database import util
+from pyannote.metrics.errors.identification import IdentificationErrorAnalysis
+
+from pyannote.audio import Audio
+from pyannote.audio.pipelines import SpeakerDiarization
+
+from ..common.utils import AudioForProdigy, before_db, get_audio_spans, get_chunks
+
+
+def diff_stream(
+    source: Path,
+    reference: dict,
+    hypothesis: dict,
+    listerrors: List,
+    diarization: bool = False,
+    chunk: float = 30.0,
+    minduration: int = 200,
+) -> Iterable[Dict]:
+
+    if os.path.isdir(source):
+        files = AudioLoader(source)
+    else:
+        name = os.path.basename(source).rsplit(".", 1)[0]
```

Comment:
Looks strangely similar to some code in `common.utils.py`.
Would it be possible to factorize some of this code?